File: backend/lab-resource-platform/src/main/java/com/lab/resource/service/EquipmentService.java
package com.lab.resource.service;

import com.lab.resource.entity.Equipment;

import java.util.List;
import java.util.Map;

public interface EquipmentService {

    // Save Equipment
    Equipment saveEquipment(Equipment equipment);

    // Get All Equipment
    List<Equipment> getAllEquipment();

    // Get Equipment By ID
    Equipment getEquipmentById(Long id);

    // Update Equipment
    Equipment updateEquipment(Long id, Equipment equipment);

    // Delete Equipment
    void deleteEquipment(Long id);

    // Equipment Utilization Analytics
    List<Map<String, Object>> getEquipmentUtilization();
}

File: frontend/src/services/equipmentService.js
import api from "./api";

// =======================
// Get all equipment
// =======================
export const getAllEquipment = async () => {

    const response = await api.get("/equipment");

    return response.data;
};

// =======================
// Get equipment by ID
// =======================
export const getEquipmentById = async (id) => {

    const response = await api.get(`/equipment/${id}`);

    return response.data;
};

// =======================
// Add new equipment
// =======================
export const addEquipment = async (equipment) => {

    const response = await api.post("/equipment", equipment);

    return response.data;
};

// =======================
// Update equipment
// =======================
export const updateEquipment = async (id, equipment) => {

    const response = await api.put(`/equipment/${id}`, equipment);

    return response.data;
};

// =======================
// Delete equipment
// =======================
export const deleteEquipment = async (id) => {

    const response = await api.delete(`/equipment/${id}`);

    return response.data;
};

// =======================
// Equipment Utilization Analytics
// =======================
export const getEquipmentUtilization = async () => {

    const response = await api.get("/equipment/utilization");

    return response.data;
};